### PR TITLE
Add map_bash_info to fields 

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Infrastructure "Add map_bash_info to fields "```

#### Purpose of change
Add map_bash_info to fields 
Make gum web bashable

#### Testing
Spawn some gum spiders
Bash the webs

### CODE REVIEW

File: src/handle_action.cpp [hunk 1]
```diff
@@ -674,6 +674,25 @@ static void smash()
             crit->use_mech_power( -3 );
         }
     }
+    for( std::pair<const field_type_id, field_entry> &fd_to_smsh : m.field_at( smashp ) ) {
+        const map_bash_info &bash_info = fd_to_smsh.first->bash_info;
+        if( bash_info.str_min == -1 ) {
+            continue;
+        }
+        if( smashskill < bash_info.str_min && one_in( 10 ) ) {
+            // %s is the smashed field
```
Comment (BevapDin) [inline src/handle_action.cpp]: If this is intended to be for the translators, it must be 
```suggestion
            //~ %s is the smashed field
```

File: src/handle_action.cpp [hunk 2]
```diff
@@ -674,12 +674,28 @@ static void smash()
...
+            continue;
+        }
+        if( smashskill < bash_info.str_min && one_in( 10 ) ) {
+            add_msg( m_neutral, _( "You don't seem to be damaging the %s." ), fd_to_smsh.first->get_name() );
+            return;
+        } else if( smashskill >= rng( bash_info.str_min, bash_info.str_max ) ) {
+            sounds::sound( smashp, bash_info.sound_vol, sounds::sound_t::combat, bash_info.sound, true, "smash",
+                           "field" );
+            m.remove_field( smashp, fd_to_smsh.first );
+            g->m.spawn_items( smashp, item_group::items_from( bash_info.drop_group, calendar::turn ) );
```
Comment (BevapDin) [inline src/handle_action.cpp]:
```suggestion
            m.spawn_items( smashp, item_group::items_from( bash_info.drop_group, calendar::turn ) );
```

File: src/mapdata.h [hunk 3]
```diff
@@ -39,18 +39,25 @@ struct map_bash_info {
     int sound_vol;          // sound volume of breaking terrain/furniture
     int sound_fail_vol;     // sound volume on fail
     int collapse_radius;    // Radius of the tent supported by this tile
+    int fd_bash_move_cost = 100; // cost to bash a field
     bool destroy_only;      // Only used for destroying, not normally bashable
     bool bash_below;        // This terrain is the roof of the tile below it, try to destroy that too
     std::string drop_group; // item group of items that are dropped when the object is bashed
     translation sound;      // sound made on success ('You hear a "smash!"')
     translation sound_fail; // sound  made on fail
+    std::string field_bash_msg_succes; // message upon successfully bashing a field
```
Comment (ZhilkinSerg) [inline src/mapdata.h]:
```suggestion
    translation field_bash_msg_success; // message upon successfully bashing a field
```

File: src/handle_action.cpp [hunk 4]
```diff
@@ -674,12 +674,28 @@ static void smash()
...
+        if( smashskill < bash_info.str_min && one_in( 10 ) ) {
+            add_msg( m_neutral, _( "You don't seem to be damaging the %s." ), fd_to_smsh.first->get_name() );
+            return;
+        } else if( smashskill >= rng( bash_info.str_min, bash_info.str_max ) ) {
+            sounds::sound( smashp, bash_info.sound_vol, sounds::sound_t::combat, bash_info.sound, true, "smash",
+                           "field" );
+            m.remove_field( smashp, fd_to_smsh.first );
+            m.spawn_items( smashp, item_group::items_from( bash_info.drop_group, calendar::turn ) );
+            u.mod_moves( - bash_info.fd_bash_move_cost );
+            add_msg( m_info, _( bash_info.field_bash_msg_succes ) );
```
Comment (ZhilkinSerg) [inline src/handle_action.cpp]:
```suggestion
            add_msg( m_info, bash_info.field_bash_msg_success.translated() );
```

File: src/mapdata.cpp [hunk 5]
```diff
@@ -230,13 +231,19 @@ bool map_bash_info::load( const JsonObject &jsobj, const std::string &member, bo
...
+            furn_set = furn_str_id( j.get_string( "furn_set", "f_null" ) );
+            break;
+        case map_bash_info::terrain:
+            ter_set = ter_str_id( j.get_string( "ter_set" ) );
+            ter_set_bashed_from_above = ter_str_id( j.get_string( "ter_set_bashed_from_above",
+                                                    ter_set.c_str() ) );
+            break;
+        case map_bash_info::field:
+            assign( j, "move_cost", fd_bash_move_cost, 100 );
+            j.read( "msg_succes", field_bash_msg_succes );
```
Comment (ZhilkinSerg) [inline src/mapdata.cpp]:
```suggestion
            j.read( "msg_success", field_bash_msg_success );
```

File: data/json/field_type.json [hunk 6]
```diff
@@ -82,6 +82,14 @@
     "priority": 2,
     "phase": "solid",
     "display_items": false,
+    "bash": {
+      "str_min": 1,
+      "str_max": 3,
+      "sound_vol": 2,
+      "sound_fail_vol": 2,
+      "sound": "hsh!",
+      "msg_succes": "You brush aside some webs."
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
      "msg_success": "You brush aside some webs."
```

File: data/mods/My_Sweet_Cataclysm/sweet_field_type.json [hunk 7]
```diff
@@ -58,6 +58,15 @@
...
     "phase": "solid",
     "half_life": "2 h",
+    "bash": {
+      "str_min": 1,
+      "str_max": 3,
+      "sound_vol": 2,
+      "sound_fail_vol": 2,
+      "sound": "shwip",
+      "sound_fail": "shwomp",
+      "msg_succes": "You brush the gum web aside."
```
Comment (ZhilkinSerg) [inline data/mods/My_Sweet_Cataclysm/sweet_field_type.json]:
```suggestion
      "msg_success": "You brush the gum web aside."
```

File: doc/JSON_INFO.md [hunk 8]
```diff
@@ -2971,4 +2971,21 @@ Setting of sprite sheets. Same as `tiles-new` field in `tile_config`. Sprite fil
...
     "id": "fd_gum_web", // id of the field
     "immune_mtypes": [ "mon_spider_gum" ], // list of monster immune to this field
+    "bash": {
+      "str_min": 1, // lower bracket of bashing damage required to bash
+      "str_max": 3, // higher bracket
+      "sound_vol": 2, // noise made when succesfully bashing the field
+      "sound_fail_vol": 2, // noise made when failing to bash the field
+      "sound": "shwip", // sound on success
+      "sound_fail": "shwomp", // sound on failure
+      "msg_succes": "You brush the gum web aside.", // message on success
```
Comment (ZhilkinSerg) [inline doc/JSON_INFO.md]:
```suggestion
      "msg_success": "You brush the gum web aside.", // message on success
```

File: doc/JSON_INFO.md [hunk 9]
```diff
@@ -2971,4 +2971,21 @@ Setting of sprite sheets. Same as `tiles-new` field in `tile_config`. Sprite fil
...
+      "move_cost": 120, // how many moves it costs to succesfully bash that field (default: 100)
+      "items": [                                   // item dropped upon succesfull bashing
```
Comment (ZhilkinSerg) [inline doc/JSON_INFO.md]:
```suggestion
      "items": [                                   // item dropped upon succesful bashing
```

**Pull request merged**